api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -46,10 +57,70 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 				case FhirConstants.COMMON_SEARCH_HANDLER:
 					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
 					break;
+				case FhirConstants.HAS_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleHasAndListParam(criteria, (HasAndListParam) param.getParam()));
+					break;
 			}
 		});
 	}
 	
+	/**
+	 * Handle _has parameters that are passed in to constrain the Encounter resource on properties of
+	 * dependent resources
+	 */
+	protected void handleHasAndListParam(Criteria criteria, HasAndListParam hasAndListParam) {
+		if (hasAndListParam != null) {
+			log.debug("Handling hasAndListParam");
+			hasAndListParam.getValuesAsQueryTokens().forEach(hasOrListParam -> {
+				if (!hasOrListParam.getValuesAsQueryTokens().isEmpty()) {
+					
+					log.debug("Handling hasOrListParam");
+					// Making the assumption that any "orListParams" match everything except for the value
+					HasParam hasParam = hasOrListParam.getValuesAsQueryTokens().get(0);
+					Set<String> values = new HashSet<>();
+					hasOrListParam.getValuesAsQueryTokens().forEach(orParam -> values.add(orParam.getParameterValue()));
+					
+					log.debug("Handling hasParam = " + hasParam.getQueryParameterQualifier());
+					log.debug("With value in " + values);
+					
+					boolean handled = false;
+					
+					// Support constraining encounter resources to those that contain only certain Medication Requests
+					if (FhirConstants.MEDICATION_REQUEST.equals(hasParam.getTargetResourceType())) {
+						if (MedicationRequest.SP_ENCOUNTER.equals(hasParam.getReferenceFieldName())) {
+							if (lacksAlias(criteria, "orders")) {
+								if (Encounter.class.isAssignableFrom(typeToken.getRawType())) {
+									criteria.createAlias("orders", "orders");
+								} else {
+									if (lacksAlias(criteria, "en")) {
+										criteria.createAlias("encounters", "en");
+									}
+									criteria.createAlias("en.orders", "orders");
+								}
+							}
+							// Constrain only on non-voided Drug Orders
+							criteria.add(Restrictions.eq("orders.class", DrugOrder.class));
+							criteria.add(Restrictions.eq("orders.voided", false));
+							
+							String paramName = hasParam.getParameterName();
+							String paramValue = hasParam.getParameterValue();
+							if (MedicationRequest.SP_INTENT.equals(paramName)) {
+								if (values.contains(MedicationRequest.MedicationRequestIntent.ORDER.toCode())) {
+									// No additional constraints needed, all Orders are given/assumed intent=order
+									handled = true;
```

Comment:
There is nothing really to handle here, I just wanted to add explicit support for this parameter name/value combination.  I'm doing that by setting a flag, and then logging a warning if any other parameter combination was found.